shared/reducers/chat.js

```diff
@@ -151,7 +151,7 @@ function reducer(state: Constants.State = initialState, action: Constants.Action
       const newInbox = action.payload.inbox.map(newRow => {
         const id = newRow.get('conversationIDKey')
         const existingRow = existingRows.find(existingRow => existingRow.get('conversationIDKey') === id)
-        return existingRow || newRow
+        return existingRow ? (existingRow.teamType !== newRow.teamType ? newRow : existingRow) : newRow
```

Comment:
@chrisnojima you might want to pay special attention to this. Without this change, hitting an inbox stale action does nothing to get the conversation into the big team list, since we just return its old row. With this in play, it properly moves. Let me know if you want to do this differently though.